### PR TITLE
Fix #11: Image transformation for optimal blob detection

### LINKED ISSUE #11: Image transformation for optimal blob detection
The goal of this issue is to address the following details in regard to image transformation to spot the orange blob in a more optimal way: 

- [x] Experiment with different values in saturation, contrast and brightness
- [x] Observe how blob is spotted

### PR DESCRIPTION
**Pull request by JocelynVelarde**
## Changes made
1. Updated constants for brightness, contrast and saturation in `distance_regression.py`

## Issue ticket number and link
Refers to issue #11

### CODE REVIEW

File: Vision/distance_regression.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def main():
             print("Angle: ", angle, " degrees")
             uart.write(str(distance) + " " + str(angle) + "\n")
         else:
-            print("No blobs found")
-            uart.write("No blobs found\n")
+            print("No blob found")
+            uart.write("No blob found\n")
```
Comment (afr2903) [inline Vision/distance_regression.py:102]: Please do not farm commits 🧐

File: Vision/distance_regression.py [hunk 2]
```diff
@@ -1,99 +1,97 @@
...
+    return blobs
+
+def calculate_distance(blob):
+    """
+    Calculates the distance of the blob from the camera.
+    """
+    relative_cx = blob.cx() - FRAME_HEIGHT
+    relative_cy = blob.cy() - FRAME_WIDTH
+    magnitude_distance = math.sqrt(relative_cx**2 + relative_cy**2)
+    total_distance = 11.83*math.exp((0.0245)*magnitude_distance)
```
Comment (afr2903) [inline Vision/distance_regression.py:68]: Great refactoring!! The structure with all the functions and variable names is more readable. For this equation of the `total_distance`, I think its okay having the values obtained from the regression as numbers, as they are only being used once. 

Just add a single inline comment explaining that it is an exponential equation, and that it was calculated with real data and a regression.

File: Vision/distance_regression.py [hunk 3]
```diff
@@ -70,6 +72,9 @@ def calculate_angle(blob):
     relative_cx = blob.cx() - FRAME_HEIGHT
     relative_cy = blob.cy() - FRAME_WIDTH
     angle = math.degrees(math.atan(relative_cy/relative_cx))
+    if angle < 0:
+        angle += 180
+        return angle
     return angle
```
Comment (afr2903) [inline Vision/distance_regression.py:82]: The `return angle` inside the if is unnecessary as there is already another `return` below. Even more, it's up to you, but you can reduce these 4 lines two one with a ternary operator:
```python
return angle+180 if (angle < 0) else angle
```
Again, is your decision if you use a ternary operator or a normal `if` :)

**Review by afr2903 — APPROVED** [review]
Check the comments I left for you. Minor changes are advised.
I want to congratulate you for this week's work, your commitment was great and if you keep the hard work I think you can win the competition ;)

File: Vision/distance_regression.py (hunk 2 above)
Comment (JocelynVelarde) [reply Vision/distance_regression.py:68]: Okei great! I'll add it.

File: Vision/distance_regression.py (hunk 3 above)
Comment (JocelynVelarde) [reply Vision/distance_regression.py:82]: That looks good

**Comment by JocelynVelarde** [comment]
Before merging into main branch , branch for issue #4 must be merged into branch for issue #11. Note that for the first release branch 11 should be the most updated one.

**Review by afr2903 — APPROVED** [review]
Thanks for taking the advice! Let's merge this issue to `main` tomorrow, let me know if you need anything else for me to review.

File: Vision/distance_regression.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    while True:
+        clock.tick()
+        img = sensor.snapshot()
+        blobs = locate_blob(img)
+        if blobs:
+            distance = calculate_distance(blobs[0])
+            angle = calculate_angle(blobs[0])
+            print("Distance: ", distance, " cm")
+            print("Angle: ", angle, " degrees")
+            uart.write(str(distance) + " " + str(angle) + "\n")
```
Comment (EmilianoHFlores) [inline Vision/distance_regression.py:97]: We'll have to pay attention to Pico/ESP performance with handling string conversion. As long as it works it's fine!

File: .vscode/arduino.json
Comment (EmilianoHFlores) [inline .vscode/arduino.json:1]: Add .vscode folder to gignore as it can cause problems when cloned to other machines.

**Review by EmilianoHFlores — APPROVED** [review]
We were very happy with your and your team's workflow this week, keep it up!

**Pull request merged**